react/components/MiniCartFooter.js

```diff
@@ -8,7 +8,32 @@ import PropTypes from 'prop-types'
 import { MiniCartPropTypes } from '../propTypes'
 import minicart from '../minicart.css'
 
-const handleClickButton = () => location.assign('/checkout/#/cart')
+const handleClickButton = () => {
+  try {
+    const segment = JSON.parse(window.atob(__RUNTIME__.segmentToken))
+    const utmParams = [
+      { key: 'utm_campaign', value: segment.utm_campaign },
+      { key: 'utm_source', value: segment.utm_source },
+      { key: 'utmi_campaign', value: segment.utmi_campaign },
+    ]
+
+    const params = utmParams.reduce((acc, param) => {
+      if (param.value) {
```

Comment:
is it possible that more than 1 `param.value` exists? In this case, the params would be only concatenated in the URL, should it use an `&` if this happens?

Reply:
You could replace that `reduce()` with something like 
> Object.entries(([key,value]) => `${key}=${encodeURIComponent(value)}`).join('&')

to support `&` as well